src/smtpproto/protocol.py

```diff
@@ -100,7 +101,8 @@ def _encode_address(self, address: Union[str, Address]) -> bytes:
         except UnicodeEncodeError:
             raise SMTPProtocolViolation(
```

Comment:
It should be able to tell which one caused the error.

Reply:
Ah, good point, thank you. I've added different cases for both of these.